qiskit_dynamics/solvers/solver_classes.py

```diff
@@ -147,8 +235,79 @@ def __init__(
                                present specify as a tuple of lists of frequencies, one for
                                Hamiltonian operators and one for dissipators.
             validate: Whether or not to validate Hamiltonian operators as being Hermitian.
+
+        Raises:
+            QiskitError: If arguments concerning pulse-schedule interpretation are insufficiently
+            specified.
         """
 
+        # set pulse specific information if specified
+        self._hamiltonian_channels = None
+        self._dissipator_channels = None
+        self._all_channels = None
+        self._channel_carrier_freqs = None
+        self._dt = None
+        self._schedule_converter = None
+
+        if any([dt, channel_carrier_freqs, hamiltonian_channels, dissipator_channels]):
+            all_channels = []
```

Comment:
Would a set be better here?

Reply:
Just tried changing to using a `set` but errors popped up as some of the entries are `dict` or `list` instances, which are unhashable, and so can't be in a `set`.